File: src/62.unique-paths.go
/*
 * @lc app=leetcode id=62 lang=golang
 *
 * [62] Unique Paths
 */

// @lc code=start
func uniquePaths(m int, n int) int {
	matrix := make([][]int, m)
	for k := 0; k < m; k++ {
		matrix[k] = make([]int, n)
	}
	matrix[0][0] = 0
	for i, v := range matrix {
		for j, _ := range v {
			if i == 0 && j == 0 {
				matrix[i][j] = 1
				continue
			}
			var a, b = i - 1, j;
			var c, d = i, j - 1;
			if a < 0 {
				a = 0
			}
			if d < 0 {
				d = 0
			}
			matrix[i][j] = matrix[a][b] + matrix[c][d]
		}
	}
	return matrix[m - 1][n - 1]
}
// @lc code=end

File: src/62.unique-paths.java
/*
 * @lc app=leetcode id=62 lang=java
 *
 * [62] Unique Paths
 */

// @lc code=start
class Solution {
    public int uniquePaths(int m, int n) {
        int[][] matrix = new int[m][n];
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (i == 0 && j == 0) {
                    matrix[i][j] = 1;
                    continue;
                }
                int a = i - 1 < 0 ? 0 : i - 1;
                int b = j - 1 < 0 ? 0 : j - 1;
                matrix[i][j] = matrix[a][j] + matrix[i][b];
            }
        }
        return matrix[m - 1][n - 1];
    }
}
// @lc code=end

